server/webhook_worker.go

```diff
@@ -55,7 +80,14 @@ func (ww webhookWorker) process(msg *webhookMessage) (err error) {
 		return err
 	}
 
-	channelsSubscribed, err := ww.p.getChannelsSubscribed(v, msg.InstanceID)
+	visibilityAttribute := ""
+	if isCommentRelatedWebhook(wh) {
+		if visibilityAttribute, err = ww.getVisibilityAttribute(msg, v); err != nil {
+			return err
+		}
+	}
```

Comment:
If the comment author is not connected to Mattermost, we end up not processing the webhook event no matter what. I think there's a design gap in the security model of this feature. This is a difficult problem to solve, the case of a comment author not connected to Mattermost.

---

Let's say a subscription is set to "comment visibility must be empty". If we can't fetch the comment because the author is not connected to MM, then we can't know the visibility of the comment. And so we would then always be avoiding the webhook events with every comment made by a user that is not connected. That's not really behavior that we want.

Though we can only assume that it is a sensitive comment if we can't fetch its visibility. Otherwise the feature would create posts for sensitive comments.

@esarafianou Do you have any thoughts on this? Essentially, this feature allows subscriptions to filter out sensitive comments, but if the comment author is not connected to MM, then we have no way of knowing whether the comment is sensitive or not. Defaulting to secure avoids unwanted data leakage, but it then requires all comment authors to be connected to MM for any comments to be posted ever.

Reply:
@mickmister Seems I had missed this. Is this still relevant?

Reply:
@mickmister I have removed the extra API call for getting the comment visibility data as in the current implementation of Jira plugin we are always making an API call to get the comment event data. So, the user have to be connected in every case of comment events.

This will be fixed with the API token approach we discussed earlier. So, maybe we can proeed with this for now and it will be fixed later.

Reply:
API token approch implemented via https://github.com/mattermost/mattermost-plugin-jira/pull/1102